packages/task-list/src/TaskListItem.tsx

```diff
@@ -76,58 +79,99 @@ const TaskListItem = ({
 		<li css={{ counterIncrement: 'task-count' }}>
 			<Flex
 				as={as}
+				flexDirection={['column', 'row']}
+				justifyContent="space-between"
+				alignItems={['flex-start', 'center']}
 				className={className}
 				gap={0.75}
-				paddingY={1.5}
-				paddingX={0.75}
+				paddingX={1}
+				paddingY={1}
 				fontFamily="body"
 				color="text"
 				borderBottom
-				borderLeft
-				borderLeftWidth="xl"
 				width="100%"
 				focus
 				css={{
-					borderLeftColor: active ? boxPalette.foregroundAction : 'transparent',
+					position: 'relative',
 					textDecoration: 'none',
+					[`[${taskListItemTextDataAttr}]`]: {
+						...packs.underline,
+						color: boxPalette.foregroundAction,
+					},
+					...(status === 'doneRecently'
+						? {
+								backgroundColor: boxPalette.systemSuccessMuted,
+						  }
+						: null),
+					...(active
+						? {
+								'&:before': {
+									content: '""',
+									background: boxPalette.foregroundAction,
+									position: 'absolute',
+									top: 0,
+									bottom: 0,
+									left: 0,
+									width: tokens.borderWidth.xl,
+								},
+						  }
+						: null),
+
 					'&:hover': {
 						backgroundColor: boxPalette.backgroundShade,
+						[`[${taskListItemTextDataAttr}]`]: {
+							textDecoration: 'none',
+							color: boxPalette.foregroundText,
+						},
 					},
 				}}
 				{...props}
 			>
-				<Icon
-					size="md"
-					color={iconColor}
-					css={{
-						// Use padding to ensure the icon is aligned centered with the status label and title
-						paddingTop: '0.75rem',
-					}}
-				/>
-				<Flex as="span" flexDirection="column" gap={0.25}>
-					<Text as="span" color="muted" fontSize="xs" lineHeight="nospace">
-						{label}
-					</Text>
-					<Text
-						fontSize="md"
-						lineHeight="heading"
-						fontWeight="bold"
-						color="action"
-						css={{
-							...packs.underline,
-							...(ordered && {
-								'&:before': {
-									content: "counter(task-count) '. '",
-								},
-							}),
-						}}
-					>
-						{children}
-					</Text>
-					<Text color="muted" fontSize="xs">
-						{message}
-					</Text>
+				<Flex as="span" gap={[0, 1]}>
```

Comment:
We need to add `as="span"` to every child of Button to ensure valid html. The HTML validation tests were great here!